### PR TITLE
Migration to Gatsby v2

### PR DESCRIPTION
**Pull request by mpoiu**
Migrated to Gatsby V2, basing on this guide:
https://next.gatsbyjs.org/docs/migrating-from-v1-to-v2/

Related issue: #72

### CODE REVIEW

**Comment by fabien0102** [comment]
Seams to have some typings error on netlify build, @mpoiu do you have access to the netlify logs?

File: src/html.tsx [hunk 1]
```diff
@@ -6,33 +6,16 @@ import Helmet from "react-helmet";
...
-  }
-}
-
 interface HtmlProps {
   body: any;
   postBodyComponents: any;
   headComponents: any;
 }
 
 // Use `module.exports` to be compliante with `webpack-require` import method
```
Comment (fabien0102) [inline src/html.tsx]: You can remove the comment

File: src/pages/404.tsx [hunk 2]
```diff
@@ -1,17 +1,26 @@
 import * as React from "react";
 import { Header, Icon, Grid } from "semantic-ui-react";
+import { default as DefaultLayout } from "../components/Layout";
```
Comment (fabien0102) [inline src/pages/404.tsx]: `import DefaultLayout from "../components/Layout";` should works here

File: src/pages/about.tsx [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
-        <p>
-          For any question, I'm on <a href="https://discord.gg/2bz8EzW" target="blank">discord #reactiflux/gatsby</a>
-        </p>
-        <p>
-          For any issues, any PR are welcoming
-          <a href="https://github.com/fabien0102/gatsby-starter/issues" target="blank"> on this repository</a>
-        </p>
-      </Segment>
-    </Container>
+    <DefaultLayout location={props.location}>
```
Comment (fabien0102) [inline src/pages/about.tsx]: Why do we need to encapsulate every pages inside this `DefaultLayout`?

File: src/html.tsx (hunk 1 above)
Comment (mpoiu) [reply src/html.tsx]: Right, thanks. Just removed it.

File: src/pages/404.tsx (hunk 2 above)
Comment (mpoiu) [reply src/pages/404.tsx]: Yes, it looks better - fixed

**Comment by mpoiu** [comment]
Thanks @fabien0102. I have committed coding style fixes and will take a look at netlify build later today or tomorrow.

File: src/pages/about.tsx (hunk 3 above)
Comment (fabien0102) [reply src/pages/about.tsx]: I'm thinking about an HOC called `withLayout` to be more flexible to in this case we will have:

```ts
const AboutPage = () => <div />;
export default withLayout(AboutPage);
```

So we don't need to repeat this `location` props everytime :wink:

**Comment by fabien0102** [comment]
Deploy preview for *fabien0102-gatsby-starter* ready!

Built with commit a80766ec284588171dd623992058e87dda789b06

https://deploy-preview-73--fabien0102-gatsby-starter.netlify.com

File: src/pages/about.tsx (hunk 3 above)
Comment (mpoiu) [reply src/pages/about.tsx]: I think it makes sense. I will try it with HOC.

**Comment by mpoiu** [comment]
@fabien0102 Thank you for testing it. I have committed fixes to issue 1, 2 and 3. Also added some other small fixes.
I will check how it looks with HOC shortly.

File: src/components/Layout.tsx [hunk 4]
```diff
@@ -57,3 +57,16 @@ export default (props: DefaultLayoutProps) => {
     </Provider>
   );
 };
+
+export default Layout;
+
+export const withLayout = <P extends object>(WrappedComponent: React.ComponentType<P>) =>
```
Comment (fabien0102) [inline src/components/Layout.tsx]: Very nice 👍

**Pull request merged**